### ISSUE TITLE
[0.13.5] error: value enablePlugins is not a member of sbt.Project

### ISSUE THREAD

**Issue by betehess** [sbt/sbt#1592]
I asked about this issue at (https://github.com/ensime/ensime-sbt/issues/60) and then at https://github.com/playframework/playframework/issues/3398. Now it really looks like the issue is with sbt itself.

Can somebody try to load https://github.com/betehess/ensime-issue-60 and if it works, then try `gen-ensime`?

Note: I have the same issue with the latest build of sbt.

**Comment by betehess** [comment]
Obviously, my brain is fried after a day of debugging sbt issues in my projects at work... The StackOverflow question I was mentionning has nothing to do with the issue being reported here. I will update the ticket accordingly. Sorry for the confusion.

**Comment by fommil** [comment]
You have my sympathies at the very least. I've been there many times.

**Comment by jsuereth** [comment]
I'm investigating now.

**Comment by jsuereth** [comment]
It appears you hit this subtle issue in Ivy.  

What is happening:
- The pom.xml file for ensime-sbt, when translated to ivy.xml, is needlessly depending on all configurations of its transitive dependencies (thanks to a missing master configuration).
- This means the `provided` configuration of transitive dependencies of ensime-sbt are wired into the `compile` configuration of ensime-sbt, leading to a ton of JARs on the classpath that should not be there.
- This means that the "sbt" jars for 0.13.1 (that do not have `enablePlugins`) are added to the legitimate classpath when compiling your classpath, BEFORE the sbt jars themselves
- This means the scala compiler is loading the sbt 0.13.1 classes when compiling rather than 0.13.5, thanks to the "exclude" notion not working in Ivy.

While we have a fix:  https://github.com/sbt/sbt/pull/1586 it's not going to hit until sbt 0.13.7 (~ 1-2 months)

Basically you should never have a maven POM.xml depend on an ivy project (for now).    There are only two sbt plugins which publish to maven repositories that I know of:  
- sbt-idea
- sbt-ensime.

My recommendation, as always, is to NOT publish sbt plugins to maven repositories.  WE provide a free repo.scala-sbt.org repository as an alternative.   I'd recommend seeing if those plugins would be willing to use bintray: http://www.scala-sbt.org/0.13/docs/Bintray-For-Plugins.html

# Workarounds
- If you publish ensime-sbt locally (via an ivy local repo) things work fine
- If you convince the plugin authors to publish to bintray, as an ivy artifact
- Wait for sbt 0.13.7 with the fix to cross ivy-maven resolution.

**Comment by betehess** [comment]
Thanks @jsuereth for looking into this!

The first workaround is good enough in my case. I believe @fommil is following this discussion, so it's up to him to decide what to do about the current issue. Updating the documentation with a reference to the issue could be good enough for now.

**Comment by jsuereth** [comment]
@betehess Which documentation would you like us to change?  Today is our bug-fix grind-a-thon, so I can try to tackle that shortly.

**Comment by betehess** [comment]
@jsuereth sorry, I meant ensime to update their documentation, pointing to [your comment](https://github.com/sbt/sbt/issues/1592#issuecomment-55411007).

**Comment by fommil** [comment]
@jsuereth does your repo do snapshots? bintray is utter shit: I looked at it, fought with it and I'm not going near it again. Simpler than sonatype my ass.

**Comment by fommil** [comment]
@betehess one more workaround you might want to try. Put this in your `~/.sbt/0.13/plugins/plugins.sbt`

```
// WORKAROUND https://github.com/sbt/sbt/issues/1439
def plugin(m: ModuleID) =
  Defaults.sbtPluginExtra(m, "0.13", "2.10") excludeAll ExclusionRule("org.scala-lang")

resolvers += Resolver.sonatypeRepo("snapshots")

libraryDependencies ++= Seq(
  plugin("org.ensime" % "ensime-sbt" % "0.1.5-SNAPSHOT"),
  plugin("com.timushev.sbt" % "sbt-updates" % "0.1.6"),
  plugin("com.typesafe.sbt" % "sbt-pgp" % "0.8.2")
)
```

**Comment by jsuereth** [comment]
@fommil My repo _is_ bintray, and no, snapshots should be fully qualified names.  If sonatype allows you to publish an ivy artifact, that'd probably be sufficient.  

Note: ivy has a notion of "latest.integration" or "latest.release" that you can use instead of -SNAPSHOT to get fully qualified release numbers but convenient checkouts.  

Also, I'm releasing a 0.13.7-M1 soon that should have a fix.

**Comment by fommil** [comment]
cool, thanks @jsuereth

**Comment by grkuntzmd** [comment]
> Basically you should never have a maven POM.xml depend on an ivy project (for now). There are only two sbt plugins which publish to maven repositories that I know of:
> - sbt-idea
> - sbt-ensime.

I believe that you can add `scalastyle-sbt-plugin` to the list.

**Comment by isomarcte** [comment]
I am still getting this exact issue on sbt 0.13.7-RC1. It was my understanding that it was fixed after the 0.13.7-M1 release. Is that correct?

**Comment by flashingpumpkin** [comment]
Another workaround: Use `project/Build.scala` for your build configuration instead of `build.sbt`.

**Comment by isomarcte** [comment]
The version >= 0.13.7-M1 works for me. I just had to delete my .ivy folder and redowload.

**Comment by SethTisue** [comment]
0.13.7-RC4 fixed my ENSIME for me — thanks @jsuereth — but not until after I blew away my `~/.ivy2`, so thanks @isomarcte for that tip

**Comment by markschaake** [comment]
I'm still getting this in 0.13.7 - though I haven't tried blowing away my `~/.ivy2` cache as @SethTisue suggests. However, I have yet another workaround that we are using:

Add this to your project's `project/plugins.sbt`

``` scala
// project/plugins.sbt
dependencyOverrides += "org.scala-sbt" % "sbt" % "0.13.7"
```

The version you specify should be the same as the version you specify in `project/build.properties`

**Comment by betehess** [comment]
I am still having the issue with 0.13.7. I have tried removing my `~/.ivy2`. I think it has been working for a while but I have reverted to a locally built ensime-sbt. Sorry, I don't have time to investigate further this time.

**Comment by benhutchison** [comment]
My experience: Hit this problem using 0.13.7. After disabling all plugins in `~/.sbt/` _and_ deleting `~/.ivy2`, I was able to clear it. Cleaning ivy cache was the last separate change I made that resolved it.

**Comment by bleadof** [comment]
So actually this workaround was found by my colleague @wvanheij . I'm not saying it's going to work for everyone but it did work for me and him.
`cat ~/.sbt/0.13/plugins/plugins.sbt`:

``` scala
resolvers += Resolver.sonatypeRepo("snapshots")

addSbtPlugin("com.typesafe.sbt" % "sbt-scalariform" % "1.3.0")

addSbtPlugin("org.ensime" % "ensime-sbt" % "0.1.6-SNAPSHOT")

dependencyOverrides += "org.scala-sbt" % "sbt" % "0.13.7"
```

**Comment by wvanheij** [comment]
Or rather, I cobbled that together from various sources, one of them being comments in this very bugreport.

**Comment by eed3si9n** [comment]
Closing this since it's probably no longer an issue in sbt 1.x.
